catalogue/__init__.py

```diff
@@ -1,11 +1,21 @@
-from typing import Sequence, Any, Dict, Tuple, Callable, Optional, TypeVar, Union
+from typing import Sequence, Any, Dict, Tuple, Callable, Optional, TypeVar
+from typing import List, Union
 import inspect
 
 try:  # Python 3.8
     import importlib.metadata as importlib_metadata
 except ImportError:
     from . import _importlib_metadata as importlib_metadata  # type: ignore
 
+try:  # Python 3.10
+    from importlib.metadata import SelectableGroups  # type: ignore
+except ImportError:
+
+    class _NotImplemented:
+        pass
+
+    SelectableGroups = _NotImplemented  # type: ignore
```

Comment:
We could also ditch `_NotImplemented` and go for `None` instead.

Reply:
That's what I tried first but it doesn't work with `isinstance` because it's not a type.